posts/2021-05-17-metrics-jee9-21006-beta.adoc

```diff
@@ -0,0 +1,178 @@
+
+Starting with the Open Liberty 21.0.0.6-beta the `RequestTimingStats` MXBean will now have it's data retrieved by the MicroProfile Metrics feature for reporting on the `/metrics` (or `/metrics/vendor`) endpoint. This functionality is compatible with both the 2.X and 3.X MicroProfile Metrics features.
+
+Below is a sample output of the new request timing metrics:
+
+[source, xml]
+----
+# TYPE vendor_requestTiming_activeRequestCount gauge
+# HELP vendor_requestTiming_activeRequestCount The number of servlet requests currently running.
+vendor_requestTiming_activeRequestCount 1
+
+# TYPE vendor_requestTiming_requestCount_total counter
+# HELP vendor_requestTiming_requestCount_total The number of servlet requests since the server started.
+vendor_requestTiming_requestCount_total 3
+
+# TYPE vendor_requestTiming_hungRequestCount gauge
+# HELP vendor_requestTiming_hungRequestCount The number of servlet requests that are currently running but are hung.
+vendor_requestTiming_hungRequestCount 0
+
+# TYPE vendor_requestTiming_slowRequestCount gauge
+# HELP vendor_requestTiming_slowRequestCount The number of servlet requests that are currently running but are slow.
+vendor_requestTiming_slowRequestCount 0
+----
+
+To be able to retrieve request timing metrics from `/metrics` you will need to enable both the requestTiming-1.0 feature in combination with one of the following MicroProfile Metric features: `mpMetrics-2.0`, `mpMetrics-2.2` or `mpMetrics-2.3` or `mpMetric-3.0`.
+
+To begin recieving metrics you will need to configure the request timing thresholds for slow or hung requests. The following example demonstrates enabling request timing metrics for `mpMetrics-3.0`, with a sample threshold configuration for both slow and hung servlet requests.
+
+[source, xml]
+----
+    <featureManager>
+	    <feature>mpMetrics-3.0</feature>
+	    <feature>requestTiming-1.0</feature>
+      <!-- other features omitted for brevity -->
+    </featureManager>
+
+    <requestTiming sampleRate="1" slowRequestThreshold="10s">
+		<servletTiming
+			slowRequestThreshold="2s"
+    		        hungRequestThreshold="10s"/>
+    </requestTiming>
+----
+
+For more information regarding Request Timing with MicroProfile Metrics:
+* link:{url-prefix}/docs/latest/reference/feature/requestTiming-1.0.html[Request Timing Documentation]
+* link:{url-prefix}/docs/latest/introduction-monitoring-metrics.html[Monitoring with Metrics]
+* link:{url-prefix}/docs/latest/jmx-metrics-list.html#request-timing-stats[Request Timing MXBean Documentation]
+
+=== Try it now 
+
+To try out these features, just update your build tools to pull the Open Liberty All Beta Features package instead of the main release. The beta works with Java SE 15, Java SE 11, or Java SE 8.
```

Comment:
The beta works with **Java SE 16**, Java SE 11, or Java SE 8.

Reply:
Updated.